src/foam/u2/view/PasswordView.js

```diff
@@ -18,7 +18,85 @@
 foam.CLASS({
   package: 'foam.u2.view',
   name: 'PasswordView',
-  extends: 'foam.u2.tag.Input',
+  extends: 'foam.u2.View',
 
-  properties: [ [ 'type', 'password' ] ]
+  css: `
+    .input-field-container {
+      position: relative;
+    }
+    .input-image {
+      position: absolute;
+      width: 24px;
+      height: 24px;
+      bottom: 8px;
+      right: 6px;
+    }
+    .input-field {
+      padding-right: 30px;
+    }
+  `,
+
+  properties: [
+    'data',
+    'type',
+    {
+      name: 'visibilityIcon',
+      value: '/foam2/src/foam/u2/images/visibility.svg'
+    },
+    {
+      class: 'Boolean',
+      name: 'passwordInvisible',
+      value: true
+    },
+    {
+      class: 'Boolean',
+      name: 'passwordIcon',
+      defaultValue: false
```

Comment:
defaultValue is from FOAM1, use value: instead. However, false is already the default so not needed.